[WIP] Disable notifications when game running

duk3luk3 opened the pull request:
Notifications while playing are bad for people using full-screen and probably generally unwanted by most players.

This PR extends the notification settings dialog to add an option for notifications while ingame, allowing to choose between enable/disable/queue notifications.

Fixes #627

duk3luk3 committed:
Maybe disable notifications when game running

duk3luk3 committed:
Make ingame notifications configurable

Add a combobox to the notification settings dialog that allows to choose
between disabling, queueing, or enabling normally notifications
triggered while the player is ingame.

This will mainly be useful for players using fullscreen since the
notification popups can screw up their game by forcing alt-tab.

Wesmania commented on src/notifications/__init__.py:
```diff
@@ -21,12 +21,27 @@ def __init__(self, client):
-        return self.disabledStartup or not self.settings.enabled
+        return (
+            self.disabledStartup
+            or self.game_running and self.settings.ingame_notifications == IngameNotification.DISABLE
```
This could use a parenthesis

duk3luk3 committed:
Refactor notifications

* Centralize checking for whether notification popup can be shown
* Make lock not block GUI

duk3luk3 committed:
Refactor notification queue processing

Removed corecursion between `showEvent` and `checkEvent`.
Loop in `checkEvent` instead.

Wesmania commented:
You're getting a signoff from me on this one.

duk3luk3 commented:
There's a typo in the code I'll need to fix, but then it'll go into develop.
